scripts/check-websockets.ts

```diff
@@ -44,50 +45,83 @@ function checkIsWebSocketAlive({
 }
 
 async function checkWebSocketEndpoints(endpoints: ChainEndpoint[]): Promise<{
-  successfulEndpoints: { endpoint: ChainEndpoint; isAlive: boolean }[];
-  failedEndpoints: { endpoint: ChainEndpoint; error: string }[];
+  successfulEndpoints: ChainEndpoint[];
+  failingEndpoints: ChainEndpoint[];
 }> {
   console.log('Checking WebSocket endpoints...');
-  const successfulEndpoints: { endpoint: ChainEndpoint; isAlive: boolean }[] =
-    [];
-  const failedEndpoints: { endpoint: ChainEndpoint; error: string }[] = [];
+  const successfulEndpoints: ChainEndpoint[] = [];
+  const failingEndpoints: ChainEndpoint[] = [];
 
   // eslint-disable-next-line no-restricted-syntax
   for (const endpoint of endpoints) {
     try {
       // eslint-disable-next-line no-await-in-loop
       const isAlive = await checkIsWebSocketAlive(endpoint);
-      successfulEndpoints.push({ endpoint, isAlive });
+      if (isAlive) {
+        successfulEndpoints.push(endpoint);
+      } else {
+        failingEndpoints.push(endpoint);
+      }
     } catch (error) {
       const errorMessage =
         error instanceof Error ? error.message : String(error);
-      failedEndpoints.push({ endpoint, error: errorMessage });
+      failingEndpoints.push(endpoint);
       console.error(`Error checking ${endpoint.chainKey}: ${errorMessage}`);
     }
   }
 
   console.log('Finished checking all endpoints.');
-  return { failedEndpoints, successfulEndpoints };
+  return { failingEndpoints, successfulEndpoints };
 }
 
-function getChainsAndEndpoints(includeTestChains: boolean) {
-  const filteredChainList = includeTestChains
+function filterChainList(includeTestChains: boolean): AnyChain[] {
+  return includeTestChains
     ? chainsList
     : chainsList.filter((chain) => !chain.isTestChain);
+}
 
-  const websocketEndpoints = filteredChainList.flatMap(({ key, ws }) => {
-    if (Array.isArray(ws)) {
-      return ws.map((endpoint) => ({
-        chainKey: key,
-        ws: endpoint,
-      }));
-    }
-    return { chainKey: key, ws };
-  });
+function getChainsAndEndpoints(includeTestChains: boolean) {
+  const filteredChainList = filterChainList(includeTestChains);
+
+  const websocketEndpoints = filteredChainList.flatMap(({ key, ws }) =>
+    ws.map((endpoint) => ({
```

Comment:
Please fix the type if there are no more just strings

Reply:
what do you mean? I removed the isArray check because now it's always array